File: solutions/exercise01/bankaccount/bankaccount.go
package bankaccount

const initialBalance = 2000

type BankAccount struct {
	number  string
	balance float64
}

func New(number string) *BankAccount {
	return &BankAccount{
		number:  number,
		balance: initialBalance,
	}
}

func (account *BankAccount) GetBalance() float64 {
	return account.balance
}

func (account *BankAccount) Deposit(amount float64) {
	account.balance += amount
}

func (account *BankAccount) Withdraw(amount float64) {
	account.balance -= amount
}

File: Exercise 1/src/main/java/bank/BankAccount.java
package bank;

import login.LoginInterface;
import user.User;

public class BankAccount extends User implements LoginInterface {

    private Integer balance;

    public BankAccount(String password, String username, String firstName, String lastName) {
        super(password, username, firstName, lastName);
    }

    @Override
    public boolean validatePassword(String p) {
        return p.equals(this.getPassword());
    }

    public Integer getBalance() {
        return this.balance;
    }
    public void addBalance(Integer amount) {
        this.balance += amount;
    }
    public boolean withdrawBalance(Integer amount) {
        if ( amount > this.balance ) {
            return false;
        } else {
            this.balance -= amount;
            return true;
        }
    }
    public boolean transferBalance(Integer amount, BankAccount toAccount) {
        if ( amount > this.balance ) {
            return false;
        } else {
            this.balance -= amount;
            toAccount.addBalance(amount);
            return true;
        }
    }




}
